### DISCUSSION TITLE
How to Force Regenerate HTML Files

### DISCUSSION THREAD

**Discussion by shuuryou** [WeblateOrg/weblate#12123] [Q&A]
Hi

I use weblate to translate some HTML documents. This works more or less reliably. However, a problem I have is that when there are non-text changes (things like a class name of a div changing) in the HTML, Weblate does not regenerate the target files for components where there are such changes. If I also have changes in the text content, there is no problem.

I think this is due to Weblate converting the HTML content to something else internally and then working on that, and converting back to HTML when saving changes. I remember it being written somewhere in the documentation.

My question is: can I somehow _force_ HTML to be regenerated without doing content/text changes?

So far my workflow for this is:
- Make whatever style changes to source HTML files and commit them to git
- Delete translation HTML files from git repo, this removes the component
- Add a 0 byte HTML file and commit to git repo, this adds the component again
- Wait
- A little later Weblate will regenerate the translation HTML file. Sometimes it'll mess up picking the right strings from the built in translation memory (today I lost four hours of time due to this, having to carefully scan for mistakes and fix manually).

If anyone knows a better way to force-regenerate translation HTML files without content changes, I'd appreciate it. I searched the docs using Google and found nothing, searched GitHub issues, thinking maybe someone requested this and it was rejected, but nothing. Hence this post.

I use Weblate using the official Docker container. If there's no GUI way to do this but a command I can type on the shell, that's perfectly fine.

**Comment by nijel** [comment]
There is forced synchronization in the repository management that should do that.

**Comment by shuuryou** [comment]
Today, by chance, there was another non-content change (`defer` added to some `<script>` tags). It seems the solution is to use the [Cleanup translation files](https://docs.weblate.org/en/latest/admin/addons.html#addon-weblate-cleanup-generic) addon.

It detected the change in every HTML file and automatically fixed the translations.

**Reply by nijel** [reply]
Let's enable it by default for such formats, done in https://github.com/WeblateOrg/weblate/pull/12337
